zerver/lib/actions.py

```diff
@@ -607,6 +607,10 @@ def do_deactivate_realm(realm: Realm) -> None:
         # notice when they try to log in.
         delete_user_sessions(user)
 
+    event = dict(type="realm_deactivated", op="remove",
+                 realm_id=realm.id)
+    send_event(event, active_user_ids(realm.id))
```

Comment:
Actually, do we even need this event?  I suppose it could be useful on mobile to force-logout users (@borisyankov thoughts?).  On web, the cookies get expired by `do_deactivate_realm` and so browsers immediately get an error and reload to a login page.

Reply:
Will the auth token get deactivated too?
I think we already handle this if so.

How do we handle a login to a deactivated realm? (what data indicates it is deactivated)

Reply:
Yep; this would just make it a bit easier to 

Trying to login to a deactivated realm returns an error; on web we redirect to a special error page.  For the mobile API, you'll get this JSON error:
```
        return json_error(_("Your realm has been deactivated."),                                     
                          data={"reason": "realm deactivated"}, status=403)                          
```

You should be able to parse the `reason` part (it's not tagged for translation).

Reply:
Though @gnprice should we refactor that interface to use the custom exception style before writing software that uses that value?

Reply:
Sending an event indicating the realm has been deactivated would be easy to handle.

Reply:
OK cool.  We'll make that happen.